hikari/impl/interaction_server.py

```diff
@@ -0,0 +1,514 @@
+                charset=_UTF_8_CHARSET,
+            )
+
+        response = await self.on_interaction(body=body, signature=signature_header, timestamp=timestamp_header)
+        return aiohttp.web.Response(status=response.status_code, headers=response.headers, body=response.payload)
+
+    async def close(self) -> None:
+        """Gracefully close the server and any open connections."""
+        if not self._runner:
+            raise errors.ComponentStateConflictError("Cannot close an inactive interaction server")
+
+        event = self._get_event()
+        runner = self._runner
+        self._application_fetch_lock = None
+        self._event = None
+        self._runner = None
+        # This shutdown then cleanup ordering matters.
+        await runner.shutdown()
+        await runner.cleanup()
+        event.set()
+
+    async def join(self) -> None:
+        """Wait for the process to halt before continuing."""
+        if not self._runner:
+            raise errors.ComponentStateConflictError("Cannot wait for an inactive interaction server to join")
+
+        await self._get_event().wait()
+
+    async def on_interaction(self, body: bytes, signature: bytes, timestamp: bytes) -> interaction_server.Response:
+        """Handle an interaction received from Discord as a REST server.
+
+        !!! note
+            If this server instance is alive then this will be called internally
+            by the server but if the instance isn't alive then this may still be
+            called externally to trigger interaction dispatch.
+
+        Parameters
+        ----------
+        body : builtins.bytes
+            The interaction payload.
+        signature : builtins.bytes
+            Value of the `"X-Signature-Ed25519"` header used to verify the body.
+        timestamp : builtins.bytes
+            Value of the `"X-Signature-Timestamp"` header used to verify the body.
+
+        Returns
+        -------
+        hikari.api.interaction_server.Response
+            Instructions on how the REST server calling this should respond to
+            the interaction request.
+        """
+        verify = self._verify or await self._fetch_public_key()
+
+        if not verify(body, signature, timestamp):
+            _LOGGER.error("Received a request with an invalid signature")
+            return _Response(_BAD_REQUEST_STATUS, b"Invalid request signature")
+
+        try:
+            payload = self._loads(body.decode("utf-8"))
+            interaction_type = int(payload["type"])
+
+        except (data_binding.JSONDecodeError, ValueError, TypeError) as exc:
+            _LOGGER.error("Received a request with an invalid JSON body", exc_info=exc)
+            return _Response(_BAD_REQUEST_STATUS, b"Invalid JSON body")
+
+        except KeyError as exc:
+            _LOGGER.error("Missing 'type' field in received JSON payload", exc_info=exc)
+            return _Response(_BAD_REQUEST_STATUS, b"Missing required 'type' field in payload")
```

Comment:
Could this spam the console if Discord try to trigger this?

Reply:
Not like it should be silent about receiving a ton of invalid payloads which as far as we're concerned are from Discord themselves, it's a lot spammier if it it manages to get down to the except Exception wrapping deserialization